File: ConsoleApplication1/ClassLibrary3/DerivedClass1.cs
using System;
using ClassLibrary1;

namespace ClassLibrary3
{
    public class DerivedClass1 : BaseClass
    {
        public override void Function()
        {
            Console.WriteLine($"[IN] {nameof(DerivedClass1)}.{nameof(Function)} is called.");
            base.Function();
            Console.WriteLine($"[OUT] {nameof(DerivedClass1)}.{nameof(Function)} is called.");
        }
    }
}

File: ConsoleApplication1/ClassLibrary2/DerivedClass1.cpp
#include "stdafx.h"
#include "DerivedClass1.h"

using namespace System;

void ClassLibrary2::DerivedClass1::Function()
{
	Console::WriteLine(L"[IN] DerivedClass1::Function is called.");
	__super::Function();
	Console::WriteLine(L"[OUT] DerivedClass1::Function is called.");
}
